pkgs/development/compilers/smlnj/default.nix

```diff
@@ -34,7 +34,15 @@ in stdenv.mkDerivation {
   patchPhase = ''
     sed -i '/PATH=/d' config/_arch-n-opsys base/runtime/config/gen-posix-names.sh
     echo SRCARCHIVEURL="file:/$TMP" > config/srcarchiveurl
-  '';
+  '' + stdenv.lib.optionalString stdenv.isDarwin (with (import <nixpkgs> {}).pkgs.darwin; ''
```

Comment:
I think using `import <nixpkgs>` inside nixpkgs is frowned upon. I believe it could cause problems if you have multiple versions of nixkpgs laying around, e.g. a channel and master, because one will now reference the other (`<nixpkgs>` depends on the value of `NIX_PATH`).

Better would be to add `darwin` as an argument up top. Nix's extreme laziness will ensure that no darwin-related things are evaluated on linux :)